### ISSUE TITLE
x/pkgsite: Broken links on pkg.go.dev to codeberg.org repositories.

### ISSUE THREAD

**Issue by catinello** [golang/go#44865]
### What is the URL of the page with the issue?

Any page on pkg.go.dev that has links resolving to repositories on codeberg.org. That issue may affect all gitea hosted repositories.
 
### What is your user agent?

Issue is not user-agent related. 

 
 
 
### What did you do?
 

Click on a link that resolves to a file at a codeberg.org repo.

Go to https://pkg.go.dev/codeberg.org/ac/ecdh25519#section-sourcefiles and take a look at the link for the file ecdh25519.go.

It refers to https://codeberg.org/ac/ecdh25519/src/commit/8a2cf2a5c93c/ecdh25519.go

If you call that link, you'll receive a 404:

```
$ curl -I https://codeberg.org/ac/ecdh25519/src/commit/8a2cf2a5c93c/ecdh25519.go
HTTP/2 404
...
```
At first it looks like an server issue, but it is not necessarily.

The correct permalink is however https://codeberg.org/ac/ecdh25519/src/commit/8a2cf2a5c93c3fb69f0e6161f4bb82dccc3584b0/ecdh25519.go 

Please spot the difference in the given commit hash value. One (defined in the link on pkg.go.dev) is abbreviated, the other is in long notation (correct destination on codeberg.org). 
 
### What did you expect to see?
 
 I expected to reach the linked target file.
 
### What did you see instead?

404

**Comment by julieqiu** [comment]
Related: https://github.com/golang/go/issues/40477

**Comment by catinello** [comment]
I just got notified by a gitea developer that the problem is known. They fixed it in the development branch and the fix will be available in the next release (1.14).

So this issue may be closed in the meantime.

https://codeberg.org/Codeberg/Community/issues/413

**Comment by julieqiu** [comment]
Thanks for letting us know, @catinello!
